### ISSUE TITLE
False positive nullability warning for generic API that's not nullable aware

### ISSUE THREAD

**Issue by Youssef1313** [dotnet/roslyn#82004]
csproj:

```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>net10.0</TargetFramework>
  </PropertyGroup>

</Project>
```

Class1.cs:

```csharp
#nullable enable
namespace ClassLibrary2
{
    public class Class1
    {
        public static void M<T>(T? p, I<T> i)
        {
            _ = i.M1(p);
            _ = i.M2(p);
        }
    }
}

#nullable disable
public interface I<in T>
{
    bool M1(T x);

    bool M2(object x);
}


```

Produces:

```
Class1.cs(8,22,8,23): warning CS8604: Possible null reference argument for parameter 'x' in 'bool I<T>.M1(T x)'.
```

I'm not expecting the warning because the API is not nullable aware. Note that only M1 is producing warning. M2 is correctly not producing any warnings.

**Comment by Youssef1313** [comment]
FYI @RikkiGibson

**Comment by RikkiGibson** [comment]
IIRC, @jcouv came up with some rules around how nullability should be adjusted when type parameters are substituted. A table maybe? I'm unsure whether we should adjust anything here, and whether we could without breaking other stuff.

Also, it feels like declaring the parameter as `I<T?> i`, would be a reasonable point of control, for saying you are fine with a nullable value going in.
